src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs

```diff
@@ -276,6 +276,9 @@ protected static T Cast<T>(object value)
             return currentDestination;
         }
 
+        private static CodeGenerationOptions ShouldGenerateMethodBody(CodeGenerationOptions options, IPropertySymbol property) =>
+            options.GenerateMethodBodies ? options.WithGenerateMethodBodies(property.IsIndexer || property.IsWriteOnly || property.IsReadOnly || property.Parameters.Length > 0) : options;
```

Comment:
Is this right in C#?  You want to generate a body instead of a new getter only autoproperty if `IsReadOnly`?